### PR TITLE
Fix taxon repository

### LINKED ISSUE #7390: [BUG] TaxonRepository::findNodesTreeSorted duplicate return.
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php#L89

### PR DESCRIPTION
**Pull request by Pitoune**
| Q               | A
| --------------- | ---
| Bug fix?        |yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #7390 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function findRootNodes()
      */
     public function findNodesTreeSorted($rootCode = null)
     {
-        return $this->createQueryBuilder('o')
+        $queryBuilder = $this->createQueryBuilder('o')
```
Comment (pamil) [inline src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php]:
```php
$queryBuilder = $this->createQueryBuilder('o')
    // ...
    ->getQuery()
    ->getResult()
;
```

Comment (Pitoune) [inline src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php]: My bad...

**Comment by pamil** [comment]
Anyway, it would be nice to write some tests for this repository, it seems to be quite tricky part of code :)

**Comment by Pitoune** [comment]
Working on it ;)

**Comment by Pitoune** [comment]
I did take a look at the tests suite and I am still wondering where to put my test.  
@pamil any recommendation ?

**Comment by Pitoune** [comment]
@pamil I finally got time to add some tests !

**Comment by pamil** [comment]
Awesome, should be much easier to add them now :)

**Comment by Pitoune** [comment]
I have added a backend ajax root to retrieve taxons from the root code. If I need it I guess some others would too.

**Comment by pamil** [comment]
Can you revert last changes and separate this bug fix from adding a new feature, please? This way we can make the repository method correct and discuss the new feature in another PR.

**Comment by Pitoune** [comment]
I am totally OK to do it but without the new feature I cannot find a way to test the `findNodesTreeSorted` method. Is that OK for you ?

**Comment by pamil** [comment]
All right, it's the first repository integration test that we want to introduce, so I can write it instead.

**Comment by Pitoune** [comment]
OK so revert is done ;)

**Comment by pjedrzejewski** [comment]
Thank you Pierre! 👍

**Pull request merged**